promotion/src/main/java/com/yas/promotion/service/PromotionService.java

```diff
@@ -0,0 +1,46 @@
+package com.yas.promotion.service;
+
+import com.yas.promotion.exception.DuplicatedException;
+import com.yas.promotion.model.Promotion;
+import com.yas.promotion.repository.PromotionRepository;
+import com.yas.promotion.utils.Constants;
+import com.yas.promotion.viewmodel.PromotionDetailVm;
+import com.yas.promotion.viewmodel.PromotionPostVm;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class PromotionService {
+    private final PromotionRepository promotionRepository;
+
+    public PromotionDetailVm create(PromotionPostVm promotionPostVm) {
+        validateIfPromotionExistedSlug(promotionPostVm.slug());
```

Comment:
You can use this annotation to avoid duplicated value @Column(unique=true) in your model, learn more 
https://www.baeldung.com/jpa-unique-constraints